download/download.go

```diff
@@ -300,16 +300,21 @@ func (d *Downloader) download(ctx context.Context, m metrics.Metrics) (*download
 			baseURL := path.Join(d.client.Config().URL, d.path)
 
 			var loader bundle.DirectoryLoader
+
+			var dup bytes.Buffer
+			teeDup := io.TeeReader(resp.Body, &dup)
```

Comment:
❓ Each "Tee" means one copy of the _raw_ bytes in memory, right? But we're OK with that, since the raw bytes aren't as large as the deserialized objects.

I'm just getting confused with all the `Tee`s here 😅 I wonder if we need the Tee-iness of this, or of a straightforward `io.Copy()` into some buffer, and passing that around as reader would be sufficient, too.

Reply:
>Each "Tee" means one copy of the raw bytes in memory, right? But we're OK with that, since the raw bytes aren't as large as the deserialized objects.
Exactly.

The idea behind using `tee` is to duplicate the stream so it can be used in multiple places.

Reply:
From what I understand, Tee does that, but in a special way: the tee'd copy buffers will be filled as the source buffer is read. A buffer that would be filled when it's created would behave differently, so I wonder if using Tee is warranted or could cause is trouble.

Reply:
I was somewhat surprised to see the use of Tee here. I would have expected us to the read the files into memory and then hang onto []byte for the data files that need to be lazily loaded. The current implementation isn't asymptotically worse in terms of memory/cpu usage, however, it does mean that the bundle serialization format (e.g., tar/gz) is exposed to the caller--they're not just dealing with a logical `path => []byte` map anymore. Perhaps a comment explaining why this approach was selected would help.